mirgecom/simutil.py

```diff
@@ -431,7 +445,334 @@ def max_component_norm(dcoll, fields, order=np.inf, *, dd=DD_VOLUME_ALL):
         componentwise_norms(dcoll, fields, order, dd=dd), actx)))
 
 
-def generate_and_distribute_mesh(comm, generate_mesh):
+class PartitioningError(Exception):
+    """Error tossed to indicate an error with domain decomposition."""
+
+    pass
+
+
+def geometric_mesh_partitioner(mesh, num_ranks=None, *, nranks_per_axis=None,
+                               auto_balance=False, imbalance_tolerance=.01,
+                               debug=False):
+    """Partition a mesh uniformly along the X coordinate axis.
+
+    The intent is to partition the mesh uniformly along user-specified
+    directions. In this current interation, the capability is demonstrated
+    by splitting along the X axis.
+
+    Parameters
+    ----------
+    mesh: :class:`meshmode.mesh.Mesh`
+        The serial mesh to partition
+    num_ranks: int
+        The number of partitions to make (deprecated)
+    nranks_per_axis: numpy.ndarray
+        How many partitions per specified axis.
+    auto_balance: bool
+        Indicates whether to perform automatic balancing.  If true, the
+        partitioner will try to balance the number of elements over
+        the partitions.
+    imbalance_tolerance: float
+        If *auto_balance* is True, this parameter indicates the acceptable
+        relative difference to the average number of elements per partition.
+        It defaults to balance within 1%.
+    debug: bool
+        En/disable debugging/diagnostic print reporting.
+
+    Returns
+    -------
+    elem_to_rank: numpy.ndarray
+        Array indicating the MPI rank for each element
+    """
+    debug = True
+    mesh_dimension = mesh.dim
+    if nranks_per_axis is None or num_ranks is not None:
+        from warnings import warn
+        warn("num_ranks is deprecated, use nranks_per_axis instead.")
+        num_ranks = num_ranks or 1
+        nranks_per_axis = np.ones(mesh_dimension, dtype=np.int8)
+        nranks_per_axis[0] = num_ranks
+    if len(nranks_per_axis) != mesh_dimension:
+        raise PartitioningError("nranks_per_axis must match mesh dimension.")
```

Comment:
```suggestion
        raise ValueError("nranks_per_axis must match mesh dimension.")
```
`ValueError` is OK here since it's checking user input.